### PR TITLE
Document new label group m #4742

### LINKED ISSUE #4742: Document new label group m
Also add more documentation to other labels

### PR DESCRIPTION
**Pull request by damithc**
Fixes #4742

### CODE REVIEW

**Comment by damithc** [comment]
@LowWeiLin @kanghj @wkurniawan07 have a look too.

File: devdocs/process.md [hunk 1]
```diff
@@ -214,11 +217,13 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
 * Open PR  
     * `s.ToReview`: Waiting for the review
     * `s.Ongoing` : The PR is being worked on.
```
Comment (thyageshm) [inline devdocs/process.md]: a trivial thing, but should we have this as "OnGoing" to match the standard of capitalizing both words?

File: devdocs/process.md [hunk 2]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
 * `p.Low`: Very little impact, unlikely to do in the near future.
 * `p.Zero`: Unlikely to do, ever.
 
+**Category**
+
+* Changes to _functionality_, categorized based on size
+  * `c.Enhancement`: An enhancement to an existing functionality (not big enough 
+   consider as a story).
+  * `c.Story`: A user story.
+  * `c.Epic`: A feature that is worth many user stories.
```
Comment (thyageshm) [inline devdocs/process.md]: so, we would be creating one Epic which is linked to multiple Stories. Then when all Stories are done, we close the Epic. Any future modification to the feature would be an Enhancement.

Am I right to interpret that way?

File: devdocs/process.md [hunk 3]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
...
+  * `c.Bug`
+  * `c.Task`: Other work items such as updating documentation.
+
 **Difficulty**
 
-* `d.Easy`: Minor change. No need to modify tests.
-* `d.Moderate`: Small, mostly-localized change. Usually requires changes to tests.
-* `d.Difficult`: Requires multiple, possibly non-localized changes. Requires changes to tests and possibly new tests.
-* `d.VeryDifficult`: Requires wide ranging tests, new tests and possibly, changes to the data schema.
+* `d.FirstTimers`: Easy. To do as the first issue for new developers. One developer
```
Comment (thyageshm) [inline devdocs/process.md]: Should we also say that the priority for such tasks must be Low?

File: devdocs/process.md [hunk 4]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
...
+  should not do more than one of these.
+* `d.Contributors`: Moderate difficulty. Small localized change. Can be done by contributors.
+* `d.Commtters`: More difficult issues that are better left for committers or more senior developers.
```
Comment (thyageshm) [inline devdocs/process.md]: d.Committers

File: devdocs/process.md [hunk 5]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
 * `p.Low`: Very little impact, unlikely to do in the near future.
 * `p.Zero`: Unlikely to do, ever.
 
+**Category**
+
+* Changes to _functionality_, categorized based on size
+  * `c.Enhancement`: An enhancement to an existing functionality (not big enough
```
Comment (thyageshm) [inline devdocs/process.md]: (not big enough to be considered as a story)

File: devdocs/process.md [hunk 6]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
...
 
-* `a.Admin`,`a.Scalability` etc. : Used to categorize issues based on the aspect
-  it tackles. This is useful when a developer is focusing on a specific aspect.
+Classifies the issues based on the non-functional aspect it tackles. 
+
+|Label             | Lead      | Co-Leads  | Description
+|------------------|-----------|-----------|------------------------------------
+|`a-AccessControl` |Tania      |           |Controlling access to user groups, authentication, privacy, anonymity
+|`a-CodeQuality`   |Wilson     |           |Refactorings that is mainly to improve code/design quality
+|`Concurrency`     |           |           |Things related to concurrent access, session control
```
Comment (thyageshm) [inline devdocs/process.md]: `a-Concurrency`

File: devdocs/process.md [hunk 7]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
...
 **Aspect**
 
-* `a.Admin`,`a.Scalability` etc. : Used to categorize issues based on the aspect
-  it tackles. This is useful when a developer is focusing on a specific aspect.
+Classifies the issues based on the non-functional aspect it tackles. 
+
+|Label             | Lead      | Co-Leads  | Description
+|------------------|-----------|-----------|------------------------------------
+|`a-AccessControl` |Tania      |           |Controlling access to user groups, authentication, privacy, anonymity
+|`a-CodeQuality`   |Wilson     |           |Refactorings that is mainly to improve code/design quality
```
Comment (thyageshm) [inline devdocs/process.md]: ... that _are_ mainly ...

File: devdocs/process.md [hunk 8]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
...
-  feature they belong to. This is useful when a developer is in charge of a
-  feature. 
-  Features vs Aspects: Features are primarily about functional requirements while Aspects 
-  are primarily about non-functional requirements.
+Classifies the issue based on the feature it involves.
 
-**Type**
+|Label       | Lead      | Co-Leads   | Description                             
+|------------|-----------|------------|-----------------------------------------
+|` f-Admin`  |Khanh      |            |Features used by Admin users
```
Comment (thyageshm) [inline devdocs/process.md]: We can remove the space before f-Admin

**Comment by thyageshm** [comment]
Just to summarize the modifications:

Now we have `Features` which are segments based on functionality (eg: Comments, Results, etc)
and each feature has one or more `Aspects` which are common across all features (eg: UI, testing, etc).

Each Feature and Aspect has their own leaders => The leaders will cover breadth by knowing the Aspect across all Features and depth by knowing one Feature fully.

Lastly, to determine who is incharge of an incoming issue (since we have Aspect and Feature), we use `m` label to specify which is of more importance in that issue.

File: devdocs/process.md (hunk 1 above)
Comment (wkurniawan07) [inline devdocs/process.md]: "ongoing" is actually one word on its own :p

File: devdocs/process.md [hunk 9]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
...
+|`a-CodeQuality`   |Wilson     |           |Refactorings that is mainly to improve code/design quality
+|`Concurrency`     |           |           |Things related to concurrent access, session control
+|`a-DevOps`        |Wilson     |           | CI, release management, version control, dev docs
+|`a-Docs`          |Jun Hao    |           |Website, user docs
+|`a-FaultTolerance`|           |           |Resilience to user errors, environmental problems
+|`a-Performance`   |           |           |Speed of operation
+|`a-Persistence`   |WeiLin     |           |Database layer, GAE datastore
+|`a-Scalability`   |Khanh      |           |Related to behavior at increasing loads
+|`a-Security`      |           |           |Protection from security threats
+|`a-Testing`       |Wilson     |           |
```
Comment (wkurniawan07) [inline devdocs/process.md]: Would "testing efficiency and robustness" be a good enough description?

File: devdocs/process.md [hunk 10]
```diff
@@ -227,42 +232,84 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
...
+
+This label is used to determine whether the `a-`, `f-`, or `t-` label 
+(described below) is used to determine the issue leader (i.e. the area lead 
+in charge of the issue). 
+
+* `m.Feature` : The owner of the `f-` label is the issue leader. The issue should 
+  have exactly one `f-` label. If there is no `m.` label, `m.Feature` is assumed. 
+* `m.Aspect` : The owner of the `a-` label is the issue leader. The issue should 
+  have exactly one `a-` label.
+* `m.Feature` : The owner of the `t-` label is the issue leader. The issue should
```
Comment (wkurniawan07) [inline devdocs/process.md]: `m.Tech`

File: devdocs/process.md (hunk 1 above)
Comment (thyageshm) [inline devdocs/process.md]: ooops you are right haha...

**Comment by damithc** [comment]
Tweaked as suggested. I'll proceed to merge.
@thyageshm yes, your observations are correct.

File: devdocs/process.md [hunk 11]
```diff
@@ -227,42 +232,85 @@ The meaning of issue prefixes: `s.` status, `a.` aspect, `f.` feature, `t.` type
...
-* `d.Moderate`: Small, mostly-localized change. Usually requires changes to tests.
-* `d.Difficult`: Requires multiple, possibly non-localized changes. Requires changes to tests and possibly new tests.
-* `d.VeryDifficult`: Requires wide ranging tests, new tests and possibly, changes to the data schema.
+* `d.FirstTimers`: Easy. To do as the first issue for new developers. One developer
+  should not do more than one of these. Not to be applied for issues with a priority `p.high` or above.
+* `d.Contributors`: Moderate difficulty. Small localized change. Can be done by contributors.
+  Not to be applied for issues with a priority `p.high` or above.
+* `d.Committers`: More difficult issues that are better left for committers or more senior developers.
+
+**Main classifier**
```
Comment (kanghj) [inline devdocs/process.md]: In line 207, m stands for 'main owner', but here it says 'Main classifier'

Comment (damithc) [inline devdocs/process.md]: Good spot. I'll fix it in the master branch.

**Pull request merged**